src/app/component/general/eco-news/news-list/news-list.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit, Input, OnDestroy } from '@angular/core';
+import { EcoNewsService } from 'src/app/service/eco-news/eco-news.service';
+import { Subscription } from 'rxjs';
+import { EcoNewsModel } from '../../../../model/eco-news/eco-news-model';
+
+@Component({
+  selector: 'app-news-list',
+  templateUrl: './news-list.component.html',
+  styleUrls: ['./news-list.component.css']
+})
+export class NewsListComponent implements OnInit, OnDestroy {
+  private view: boolean;
+  private iterator: number;
+  private toggler = false;
+
+  @Input() gridOutput: Array<string>;
+
+  private ecoNewsSubscription: Subscription;
+  private allEcoNews: EcoNewsModel[] = [];
+  private elements: EcoNewsModel[] = [];
+  public remaining = 0;
+
+  constructor(private ecoNewsService: EcoNewsService) { }
+
+  ngOnInit() {
+    this.fetchAllEcoNews();
+  }
+
+  private fetchAllEcoNews(): void {
+    this.ecoNewsSubscription = this.ecoNewsService
+      .getAllEcoNews()
+      .subscribe((data: EcoNewsModel[]) => this.setAllAndStartingElems(data));
```

Comment:
You can rewrite it like .subscribe(this.setAllAndStartingElems.bind(this));

Reply:
Fixed